### DISCUSSION TITLE
Make all FastAPI models `frozen`

### DISCUSSION THREAD

**Discussion by caniko** [fastapi-users/fastapi-users#1380] [Ideas]
Hello, I think we should make all the Pydantic models that are used towards FastAPI response models `frozen`. I ask for this feature primarily because I would like the static typechecker to stop me from mutating response models, and to make the response models hashable.

I actually would benefit quite heavily from the hashability as I put some model instances into `set` HashSet.

An alternative solution is for me to make my own class implementations `frozen=True`, [this](https://github.com/pydantic/pydantic/issues/9171) is currently impossible without pyright complaining (which is fine).

**Comment by frankie567** [comment]
I guess we're talking about schemas like `BearerResponse` or `OAuth2AuthorizeResponse`, right? If so, yes, I'm not against it.

For users schema, it should be quite easy to enable it like this in your app:

```py
class UserRead(schemas.BaseUser[uuid.UUID]):
    model_config = ConfigDict(frozen=True)
```

[Pydantic merges parent config](https://docs.pydantic.dev/latest/concepts/config/), so you are also sure to keep the settings set by `BaseUser`.

**Reply by caniko** [reply]
Yeah, that is what I am doing. Just suggesting doing this at the package level.
